src/main/java/org/example/bookstore/exception/CustomGlobalExceptionHandler.java

```diff
@@ -45,6 +45,13 @@ protected ResponseEntity<Object> handleEntityNotFoundException(EntityNotFoundExc
         return new ResponseEntity<>(body, status);
     }
 
+    @ExceptionHandler(EmptyCartException.class)
+    protected ResponseEntity<Object> handleEmptyCartException(EmptyCartException ex) {
```

Comment:
you have numerous identical bad request handlers, combine them into 1
```suggestion
    @ExceptionHandler({
        OtherBadRequestException.class,
        EmptyCartException.class})
    protected ResponseEntity<Object> handleBadRequestException(RuntimeException ex) {
        // common handling logic
    }
```